ios/FluentUI/Vnext/List/List.swift

```diff
@@ -70,12 +70,32 @@ public struct MSFListView: View {
                           with: windowProvider)
     }
 
+    /// Finds the last cell directly adjacent to the end of a list section. This is used to remove the redundant separator that is
+    /// inserted between each cell. Used as a fix until we are able to use the new List Separators in iOS 15.
+    private func findLastCell(_ lastCell: MSFListCellState) -> MSFListCellState {
+        if let children = lastCell.children {
+            if lastCell.isExpanded {
+                guard let lastChild = children.last else {
+                    preconditionFailure("Does not contain any children cells")
+                }
+                return findLastCell(lastChild)
+            } else {
+                return lastCell
+            }
+        } else {
+            return lastCell
+        }
+    }
+
     private func updateCellDividers() -> [MSFListSectionState] {
         state.sections.forEach { section in
             section.cells.forEach { cell in
                 cell.hasDivider = section.hasDividers
             }
-            section.cells.last?.hasDivider = false
+            if let lastCell = section.cells.last {
```

Comment:
also don't we only need to go through this traversing when this section has dividers right? if it is a list view which no divider we don't need to do this calculation and skip?